python/shotgun_data/shotgun_data_retriever.py

```diff
@@ -679,22 +787,35 @@ def _get_thumbnail_path(url, bundle, directory_only=False):
         # sharding methodology, see 
         # http://stackoverflow.com/questions/13841931/using-guids-as-folder-names-splitting-up
         #
-        # From the hash, generate paths on the form C1C2/C3C4/rest_of_hash.jpeg
-        # (where C1 is the first character of the hash.)
-        # for a million evenly distributed items, this means ~15 items per folder
+        # From the hash, generate paths on the form C1C2/C3C4/rest_of_hash.ext
+        # (where C1 is the first character of the hash). For a million evenly distributed
+        # items, this means ~15 items per folder
         first_folder = hash_str[0:2]
         second_folder = hash_str[2:4]
-        path_chunks = [first_folder, second_folder]
+
+        # Establish the cache path location
+        cache_path_items = [
+            bundle.cache_location, "thumbs", first_folder, second_folder
+        ]
 
         # If we were only asked to give back a directory path then we can
         # skip building and appending a file name.
         if not directory_only:
-            path_chunks.append("%s.jpeg" % hash_str[4:])
+            path_base = hash_str[4:]
+            url_base, url_ext = os.path.splitext(url_obj.path)
```

Comment:
please add a comments to help readers understand what these to returned chunks would typically contain